test/session.test.js

```diff
@@ -59,6 +59,228 @@ describe(`session: ${printPath("[test/session.test.js]")}`, function () {
         await cleanST();
     });
 
+    //- check if output headers and set cookies for create session is fine
+    it("test that output headers and set cookie for create session is fine", async function () {
+        await startST();
+        SuperTokens.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [Session.init()],
+        });
+
+        const app = express();
+
+        app.use(SuperTokens.middleware());
+
+        app.post("/create", async (req, res) => {
+            await Session.createNewSession(res, "", {}, {});
+            res.status(200).send("");
+        });
+
+        app.use(SuperTokens.errorHandler());
+
+        let res = await new Promise((resolve) =>
+            request(app)
+                .post("/create")
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+        assert(res.header["front-token"] !== undefined);
+        assert(res.header["access-control-expose-headers"] === "front-token, id-refresh-token, anti-csrf");
+        assert(res.header["id-refresh-token"] !== undefined);
+        assert(res.header["anti-csrf"] !== undefined);
+    });
+
+    //- check if output headers and set cookies for refresh session is fine
+    it("test that output headers and set cookie for refresh session is fine", async function () {
+        await startST();
+        SuperTokens.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [Session.init()],
+        });
+
+        const app = express();
+        app.use(SuperTokens.middleware());
+
+        app.post("/create", async (req, res) => {
+            await Session.createNewSession(res, "", {}, {});
+            res.status(200).send("");
+        });
+
+        app.post("/auth/session/refresh", async (req, res) => {
+            await Session.refreshSession(req, res);
+            res.status(200).send("");
+        });
```

Comment:
No need to add the refresh API as the middleware should add it automatically. In fact, this API should not even have been called. The fact that you have this means you did not test for failure condition...